Dockerfile

```diff
@@ -1,12 +1,27 @@
-ARG SOURCE_BINARY=bin/manager
+FROM golang:1.17 as builder
+
+ARG TAG
+ARG COMMIT
+ARG REPO_INFO
+WORKDIR /workspace
+
+ADD go.mod .
+ADD go.sum .
+RUN go mod download
+
+COPY pkg/ pkg/
+COPY api/ api/
+COPY cmd/ cmd/
+COPY controllers/ controllers/
+COPY main.go main.go
+# Build
+RUN CGO_ENABLED=0 GOOS=linux GO111MODULE=on go build -a -o manager main.go
 
 # Use distroless as minimal base image to package the manager binary
 # Refer to https://github.com/GoogleContainerTools/distroless for more details
-FROM gcr.io/distroless/static:nonroot
+FROM gcr.io/distroless/static:nonroot AS distroless
 WORKDIR /
-COPY ${SOURCE_BINARY} .
-EXPOSE 8080 8443
```

Comment:
We need this for kube-proxy right?

Reply:
Those are default port. Addon-mgr does not really use them. Yeah, good observation